pandas/_libs/tslibs/conversion.pyx

```diff
@@ -355,6 +366,26 @@ cdef _TSObject convert_datetime_to_tsobject(datetime ts, object tz,
         obj.value += nanos
         obj.dts.ps = nanos * 1000
 
+    if tz is not None:
+        if is_utc(tz):
```

Comment:
Can you combine as `is_utc(tz) or  is_tzlocal(tz)`?

Reply:
Done.
Wanted to emphasize that we might want to implement something for is_tzlocal in the future, but the TODO is clear enough.